### PR TITLE
fix: update dependencies

### PR DESCRIPTION
**Pull request by himkt**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Hi, @BastianZim, thank you for maintaining konoha-feedstock. I'm author of konoha package.
I just found dependencies described in `recipe/meta.yml` is bit outdated. This PR fixes them.

### CODE REVIEW

**Comment by himkt** [comment]
@conda-forge-admin, please rerender

**Comment by himkt** [comment]
And one question: can I join konoha-feedstock as a maintainer?
I'm already a member of conda-forge and I'd be happy if I can help you to publish the package.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,32 +11,33 @@ source:
...
-    - sphinx_rtd_theme >=0.4.3,<0.5.0
+    - sudachipy >=0.5.0,<0.6.0
+    - boto3 >=1.34.0,<2.0.0
+    - fastapi <1.0.0
+    - uvicorn <0.26.0
+    - sudachidict-core 20230927
+    - sphinx >=5.1.1,<6.0.0
+    - pydata-sphinx-theme >=0.10.1,<0.11.0
     - nagisa >=0.2.7,<0.3.0
+    - rich >=13.7.0,<14.0.0
```
Comment (himkt) [inline recipe/meta.yaml]: Generated using grayskull (`grayskull pypi konoha`).

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,32 +11,33 @@ source:
   sha256: 5fe59118c7fe5373ae08a2360abb6f2a6891b6eb4235b76b3461f0427b5b29b0
 
 build:
-  number: 0
+  number: 1
```
Comment (himkt) [inline recipe/meta.yaml:14]: bumped by hand.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,32 +11,33 @@ source:
...
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
-    - pip
+    - python >=3.8,<4.0
     - poetry >=1.0.2
-    - python >=3.8
+    - poetry-core
```
Comment (BastianZim) [inline recipe/meta.yaml]: Upstream you have poetry>=1.0.2. Can you please verify which one is correct?

File: recipe/meta.yaml (hunk 1 above)
Comment (BastianZim) [reply recipe/meta.yaml]: Looks great, thank you. Did you also manually verify if it is correct? Grayskul is not always 100%.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Hi @himkt thanks for opening the PR and of course, please feel free to add yourself as a maintainer. 
Just in this PR should be fine. 
Thank you!

File: recipe/meta.yaml (hunk 3 above)
Comment (himkt) [reply recipe/meta.yaml]: Thank you, I think you're right. Just removed (only with poetry) 47d4284

File: recipe/meta.yaml (hunk 1 above)
Comment (himkt) [reply recipe/meta.yaml]: Sorry for the 🔰 question but how should we verify?

At least, I manually checked `pyproject.toml` on upstream and `meta.yaml` and these were aligned after 47d4284. 📝

Comment (BastianZim) [reply recipe/meta.yaml]: Oh yeah that’s all I meant, just a quick double check. 😂

**Comment by BastianZim** [comment]
Can you just add yourself as a maintainer and rerender please and I’ll merge it.

**Comment by himkt** [comment]
> Hi @himkt thanks for opening the PR and of course, please feel free to add yourself as a maintainer.
> Just in this PR should be fine.

I read the documentation as well... 👀 

> Contributing a package to conda-forge makes you the maintainer of that package` in the documentation

I see! I now udnerstand what you said (I forgot how I became the maintainer of another repository 😄 ), thank you!

**Comment by himkt** [comment]
@conda-forge-admin, please rerender

**Comment by BastianZim** [comment]
Ahh, yeah, there are multiple ways. You can also use the bot command but the only thing that does is opening a PR like this one so both in one is easer.

**Comment by BastianZim** [comment]
@conda-forge-admin, please restart ci

**Comment by BastianZim** [comment]
Hmm something gave the CI a hickup.

**Pull request merged**